ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/verify_devices_on_cv.py

```diff
@@ -22,11 +24,59 @@ async def verify_devices_on_cv(devices: list[CVDevice], workspace_id: str, skip_
     if not devices:
         return
 
+    # Finding entries with same serialNum or same mac address
+    sysMacToHostname = {}
+    serialNumToHostname = {}
+    for e in devices:
+        if e.system_mac_address:
+            if e.system_mac_address in sysMacToHostname:
+                sysMacToHostname[e.system_mac_address].append(e.hostname)
+            else:
+                sysMacToHostname[e.system_mac_address] = [e.hostname]
+
+        if e.serial_number:
+            if e.serial_number in serialNumToHostname:
+                serialNumToHostname[e.serial_number].append(e.hostname)
+            else:
+                serialNumToHostname[e.serial_number] = [e.hostname]
+    hostnamesSameMac = [hostnames for mac, hostnames in sysMacToHostname.items() if len(hostnames) > 1]
```

Comment:
Maybe a list is not the best form inside the dict. A set would be better, since we should allow multiple entries with the same information.